src/generic/find.h

```diff
@@ -0,0 +1,37 @@
+
+namespace simdutf {
+namespace SIMDUTF_IMPLEMENTATION {
+namespace {
+namespace util {
+
+simdutf_really_inline const char *find(const char *start, const char *end,
+                                       char character) noexcept {
+  for (; std::distance(start, end) >= 64; start += 64) {
```

Comment:
FYI: Std::distance has a O(N) complexity avoid calling it will improve performance.

Reply:
I believe that, in this instance, it is constant time because a pointer is a "LegacyRandomAccessIterator".

https://en.cppreference.com/w/cpp/named_req/RandomAccessIterator

<img width="827" alt="Capture d’écran, le 2025-05-20 à 23 35 50" src="https://github.com/user-attachments/assets/3e924add-4cd2-4ad0-9399-343e39f44d94" />


I believe that it is even applies to cases like this...

```C++
std::vector<X> v =....
return std::distance(v.begin(), v.end());
```


It is possible that an alternative coding could improve performance in some cases. I am not sure.